gui/src/vmm/hv/linux/x86_64.rs

```diff
@@ -262,6 +262,106 @@ impl<'a> CpuStates for KvmStates<'a> {
     fn get_st7(&mut self) -> Result<[u8; 10], Self::Err> {
         Ok(self.fregs.fpr[7][..10].try_into().unwrap())
     }
+
+    fn get_fcw(&mut self) -> Result<u32, Self::Err> {
+        Ok(self.fregs.fcw.into())
+    }
+
+    fn get_fsw(&mut self) -> Result<u32, Self::Err> {
+        Ok(self.fregs.fsw.into())
+    }
+
+    fn get_ftwx(&mut self) -> Result<u32, Self::Err> {
+        Ok(self.fregs.ftwx.into())
+    }
+
+    fn get_fiseg(&mut self) -> Result<u32, Self::Err> {
+        Ok((self.fregs.last_ip >> 32) as u32)
+    }
+
+    fn get_fioff(&mut self) -> Result<u32, Self::Err> {
+        Ok((self.fregs.last_ip & 0xFFFFFFFF) as u32)
+    }
+
+    fn get_foseg(&mut self) -> Result<u32, Self::Err> {
+        Ok((self.fregs.last_dp >> 32) as u32)
+    }
+
+    fn get_fooff(&mut self) -> Result<u32, Self::Err> {
+        Ok((self.fregs.last_dp & 0xFFFFFFFF) as u32)
+    }
```

Comment:
This ones looks weird. Should `seg` part come from the lower words instead of upper words?

Reply:
Swapped

Reply:
I'm not sure if my guess is correct. If you know your original version is a correct one then we should go with it.

Reply:
![image](https://github.com/user-attachments/assets/1e4fe745-4c2b-4ebd-b611-70a7ddf33d7b)
![image](https://github.com/user-attachments/assets/0e8f9de2-618f-4f2d-970e-a6775915b279)